eth_defi/aave_v3/events.py

```diff
@@ -0,0 +1,311 @@
+        "log_index": int(log["logIndex"], 16),
+    }
+
+
+def decode_reserve_data_updated(aave_network_name: str, log: LogResult) -> dict:
+    """Process a ReserveDataUpdated event. The event signature is:
+
+    .. code-block::
+
+        # topic0: signature 0x804c9b842b2748a22bb64b345453a3de7ca54a6ca45ce00d415894979e22897a
+        event ReserveDataUpdated(
+            address indexed reserve,    # topic1
+            uint256 liquidityRate,      # data0
+            uint256 stableBorrowRate,   # data1
+            uint256 variableBorrowRate, # data2
+            uint256 liquidityIndex,     # data3
+            uint256 variableBorrowIndex # data4
+        );
+    """
+    # Ensure the event comes from the correct smart contract
+    if log["address"].lower() != AAVE_V3_NETWORKS[aave_network_name.lower()].pool_address.lower():
+        return None
+
+    # Do additional lookup for the token data
+    # web3 = log["event"].web3
+    # token_cache: TokenCache = log["context"]
+    result = _decode_base(log)
+
+    # Any indexed Solidity event parameter will be in topics data.
+    # The first topics (0) is always the event signature.
+    if len(log["topics"]) < 2:
+        logging.warn(f'IGNORING EVENT: block={log["blockNumber"]} tx={log["transactionHash"]} topics={log["topics"]} data={log["data"]}')
+        return None
+    event_signature, reserve = log["topics"]
+    deposit_address = convert_uint256_string_to_address(reserve)
+
+    # Any non-indexed Solidity event parameter will be in the data section.
+    # Chop data blob to byte32 entries
+    liquidity_rate_raw, stable_borrow_rate_raw, variable_borrow_rate_raw, liquidity_index_raw, variable_borrow_index_raw = decode_data(log["data"])
+
+    liquidity_rate = convert_int256_bytes_to_int(liquidity_rate_raw)
+    stable_borrow_rate = convert_int256_bytes_to_int(stable_borrow_rate_raw)
+    variable_borrow_rate = convert_int256_bytes_to_int(variable_borrow_rate_raw)
+    liquidity_index = convert_int256_bytes_to_int(liquidity_index_raw)
+    variable_borrow_index = convert_int256_bytes_to_int(variable_borrow_index_raw)
+
+    result.update({
+        "reserve": deposit_address,
+        "liquidity_rate": liquidity_rate,
+        "stable_borrow_rate": stable_borrow_rate,
+        "variable_borrow_rate": variable_borrow_rate,
+        "liquidity_index": liquidity_index,
+        "variable_borrow_index": variable_borrow_index,
+        "contract_address": log["address"],
+    })
+
+    # Detect token name from reserve address (None if not found)
+    result["token"] = aave_v3_get_token_name_by_deposit_address(deposit_address)
+
+    logger.debug(f'EVENT: block={log["blockNumber"]} tx={log["transactionHash"]} token={result["token"]} reserve={deposit_address} liquidity_rate={liquidity_rate} stable_borrow_rate={stable_borrow_rate} variable_borrow_rate={variable_borrow_rate} liquidity_index={liquidity_index} variable_borrow_index={variable_borrow_rate}')
```

Comment:
(nit)
Some of the `key=value` items could be shortened considerably by appending `=` to them and f-strings will do the rest. For example, instead of writing:
```py
f" ...  liquidity_index={liquidity_index}  ..."
```
we could write
```py
f" ...  {liquidity_index=}  ..."
```